default/python/AI/PolicyAI.py

```diff
@@ -196,18 +232,38 @@ def _unlocked_by(name: str) -> Optional[str]:
             return diversity
         return None
 
+    _rating_functions = {
+        propaganda: lambda self: 20 + self._rate_opinion(propaganda),
+        algo_research: lambda self: self._rate_algo_research(),
+        diversity: lambda self: self._rate_diversity(),
+        artisans: lambda self: self._rate_artisans(),
+        # military policies are mostly chosen by opinion, plus some rule-of-thumb values
+        allied_repair: lambda self: self._rate_opinion(charge),  # no effect, unless we have allies...
+        charge: lambda self: 5 + self._rate_opinion(charge),  # A small bonus in battle
+        scanning: lambda self: 20 + self._rate_opinion(scanning),  # May help us detect ships and planets
+        simplicity: lambda self: 20 + self._rate_opinion(simplicity),  # Makes simple ships cheaper
+        engineering: lambda self: self._rate_engineering_corps(),
+        exploration: lambda self: 10 + self._rate_opinion(exploration),  # may give a little research, speeds up scouts
+        flanking: lambda self: 5 + self._rate_opinion(flanking),  # A small bonus in battle
+        recruitment: lambda self: 15 + self._rate_opinion(recruitment),  # cheaper troop ships
+    }
+
     @cache_for_current_turn
     def _rate_policy(self, name: str) -> float:
         """Gives a rough value of a policy."""
-        if name == propaganda:
-            return 20 + self._rate_opinion(name)
-        if name == algo_research:
-            return self._rate_algo_research()
-        if name == diversity:
-            return self._rate_diversity()
-        if name == artisans:
-            return self._rate_artisans()
-        raise (ValueError(f"_rate_policy {name} not yet supported"))
+        f = self._rating_functions.get(name)
+        if assertion_fails(f, f"_rate_policy({name}) not yet supported"):
+            return 0.0
+        return f(self)
+
+    def _rate_costs(self, costs: float) -> float:
+        """
+        How do we rate the costs in comparison to a policies rating?
+        Generally costs / 3, but less if we have plenty if IP.
+        """
+        unreserved_ip = self._ip - self._wanted_ip
+        cost_or_prio = (costs + self._aistate.get_priority(PriorityType.RESOURCE_INFLUENCE)) / 2
```

Comment:
`cost_or_prio`   not sure what you mean by or in that name.

Reply:
I had problems coming up with a good name, but actually when looking at it now, the formula made not sense anyway. Rewrote it and added a comment. It's probably one of the places that may be tweaked again.